src/System.IO.FileSystem.Watcher.Polling/System/IO/PollingFileSystemWatcher.cs

```diff
@@ -14,82 +11,73 @@
-        /// <param name="includeSubdirectories">A bool controlling whether or not subdirectories will be watched too</param>
-        /// <param name="pollingIntervalInMilliseconds">Polling interval</param>
-        public PollingWatcher(string directory, bool includeSubdirectories = false, int pollingIntervalInMilliseconds = 1000)
+        /// <param name="filter">The type of files to watch. For example, "*.txt" watches for changes to all text files.</param>
+        public PollingFileSystemWatcher(string directory, string filter = "*.*")
         {
-            if (!Directory.Exists(directory))
+            if (!IO.Directory.Exists(directory))
                 throw new ArgumentException("Directory not found.", nameof(directory));
 
-            Tracing = new TraceSource("PollingWatcher");
-            _state = new PathToFileStateHashtable(Tracing);
-            _pollingIntervalInMilliseconds = pollingIntervalInMilliseconds;
-            _includeSubdirectories = includeSubdirectories;
-            _directory = directory;
+            _state = new PathToFileStateHashtable();
+            Directory = directory;
+            Filter = filter;
         }
 
         /// <summary>
-        /// Extensions to watch
+        /// Creates an instance of a watcher
         /// </summary>
-        /// <param name="extension">for example "txt", "doc", etc., i.e don't use wildcards</param>
-        /// <remarks>
-        /// By default all extensions are watched. Once this method is called, only extensions in added are watched.
-        /// </remarks>
-        public void AddExtension(string extension)
+        /// <param name="directory">The directory to watch. It does not support UNC paths (yet).</param>
+        /// <param name="includeSubdirectories">A bool controlling whether or not subdirectories will be watched too</param>
+        /// <param name="pollingIntervalInMilliseconds">Polling interval</param>
+        public PollingFileSystemWatcher(string directory, bool includeSubdirectories = false, int pollingIntervalInMilliseconds = 1000)
         {
-            if (_timer != null)
-            {
-                throw new InvalidOperationException(Strings.InvalidOperation_Extension);
-            }
-            if (_extensionsToWatch == null)
-            {
-                _extensionsToWatch = new List<string>();
+            if (!IO.Directory.Exists(directory))
+                throw new ArgumentException("Directory not found.", nameof(directory));
+
+            _state = new PathToFileStateHashtable();
+            PollingIntervalInMilliseconds = pollingIntervalInMilliseconds;
+            IncludeSubdirectories = includeSubdirectories;
+            Directory = directory;
             }
-            _extensionsToWatch.Add($"*{extension}");
-        }
+
+        public string Filter { get; set; } = "*.*";
+        public bool IncludeSubdirectories { get; set; } = false;
```

Comment:
I'd dump this and switch to taking System.IO.EnumerationOptions (and passing it through to your custom enumerator).

Reply:
What are your thoughts on the [constructors](https://github.com/dotnet/corefxlab/pull/2235/files/600bc7d86e6313f8ca698b5abfbd29d346830889#diff-bf9f0af07091216198ca86b115fd2777)? The first one `ctor(string, string)` reflects FSW. `ctor(string, bool, int)` was already there. The latter is passing in `includeSubdirectories` which can be replaced with `EnumerationOptions` and `pollingIntervalInMilliseconds` doesn't seem common enough. I think we could get by with one constructor `PollingFileSystemWatcher(string path, string filter = "*.*", EnumerationOptions options = null)`

Reply:
I prefer the one constructor.